packages/compiler-cli/ngcc/test/migrations/undecorated_parent_migration_spec.ts

```
@@ -204,38 +204,6 @@ runInEachFileSystem(() => {
          expect(decorator.import).toEqual({from: '@angular/core', name: 'Directive'});
          expect(decorator.args !.length).toEqual(0);
        });
-
-    it('should skip the base class if it is in a different package from the derived class', () => {
```

I deleted this test because it's being caught as a compile-time error now.

---

I can't remember exactly but I thought we were trying to avoid errors such as this in ngcc - i.e. it should be more tolerant than ngtsc? But perhaps this is not one of those cases?

---

I haven't been part of those discussions, but this is a bit of a weird case because the error contradicts the migration completely. Shouldn't the migrations be run before the code even gets to the compiler?

---

Migrations are run after the "analyze" phase but before "resolve" and "compile" phases.

---

I can see that this error is being identified in the "resolve" phase, so I would expect this migration to avoid the error from appearing??

---

I think the scenario is still valid to test. Here's what I think should happen in the context of ngcc:

When compiling `BaseClass` in its separate package, there is zero indication that it's an Angular class, so no action is taken against it and no error is produced.

When compiling `DerivedClass`, then, ngcc will attempt to apply the migration for undecorated parents. It will follow `DerivedClass`' inheritance hierarchy, recognize that `BaseClass` would need to be migrated but cannot be (because it's in a separate package).

At this point, ngcc has a choice:
1) raise an error, because there's no hope this would work in production and no possibility of continuing.
2) silently ignore the problem and allow the crash to happen at runtime.

Currently we use door #2, but this PR I believe will convert us to door #1. That's fine (I think the error message will be better, for sure), but we should definitely have a test for this scenario. I would also really like to see us have an error message that makes sense in the context of ngcc ("Cannot generate a factory for 'DerivedClass' from 'test-package' as it inherits from the undecorated class 'BaseClass' from 'other-package'.").